packages/arb-token-bridge-ui/src/components/TransactionHistory/TransactionsTable/TransactionsTableCctp.tsx

```diff
@@ -0,0 +1,151 @@
+  const { transfers, depositIds, withdrawalIds } = useCctpState()
+  const {
+    depositsError,
+    withdrawalsError,
+    isLoadingDeposits,
+    isLoadingWithdrawals
+  } = useCctpFetching({
+    l1ChainId: l1.network.id,
+    walletAddress: address,
+    pageSize: pageParams.pageSize,
+    pageNumber: pageParams.pageNumber,
+    type
+  })
+  const isLoading =
+    type === 'deposits' ? isLoadingDeposits : isLoadingWithdrawals
+  const hasError = type === 'deposits' ? depositsError : withdrawalsError
+
+  const status = useMemo(() => {
+    if (isLoading) return TableStatus.LOADING
+    if (hasError) return TableStatus.ERROR
+    return TableStatus.SUCCESS
+  }, [hasError, isLoading])
+
+  const transactions = useMemo(() => {
+    const startIndex = pageParams.pageNumber * pageParams.pageSize
+    const ids = type === 'deposits' ? depositIds : withdrawalIds
+    return ids
+      .map(id => transfers[id])
+      .slice(
+        startIndex,
+        startIndex + pageParams.pageSize
+      ) as unknown as MergedTransaction[]
+  }, [
+    depositIds,
+    pageParams.pageNumber,
+    pageParams.pageSize,
+    transfers,
+    type,
+    withdrawalIds
+  ])
+
+  useEffect(() => {
+    setPageParams(prevPageParams => ({
+      ...prevPageParams,
+      pageNumber: 0
+    }))
+  }, [setPageParams, type])
+
+  const tabs = useMemo(() => {
+    return [
+      {
+        handleClick: () => setType('deposits'),
+        isActive: type === 'deposits',
+        text: 'Deposits'
+      },
+      {
+        handleClick: () => setType('withdrawals'),
+        isActive: type === 'withdrawals',
+        text: 'Withdrawals'
+      }
+    ]
+  }, [type])
+
+  return (
+    <>
+      {!isSmartContractWallet && <TransactionsTableSwitch tabs={tabs} />}
+
+      {/* search and pagination buttons */}
+      <TableActionHeader
+        type={'cctp'}
+        pageParams={pageParams}
+        setPageParams={setPageParams}
+        transactions={transactions}
+        isSmartContractWallet={isSmartContractWallet}
+        loading={isLoading}
+        showSearch={false}
+      />
+
+      <table className="w-full overflow-hidden rounded-b-lg bg-white">
+        <TransactionsTableHeader />
+
+        <tbody>
+          {status === TableStatus.LOADING && <TableBodyLoading />}
+
+          {status === TableStatus.ERROR && <TableBodyError />}
+
+          {/* when there are no transactions present */}
+          {status === TableStatus.SUCCESS && !transactions.length && (
+            <EmptyTableRow>
+              <span className="text-sm font-medium">No transactions</span>
+            </EmptyTableRow>
+          )}
+
+          {/* finally, when transactions are present, show rows */}
+          {status === TableStatus.SUCCESS &&
+            transactions.map((tx, index) => {
+              const isLastRow = index === transactions.length - 1
+              if (isDeposit(tx)) {
+                return (
+                  <TransactionsTableDepositRow
```

Comment:
Would it have been easier to always use a withdrawal row for CCTP? Similar to how you did with cards?